### DISCUSSION TITLE
EPS Output

### DISCUSSION THREAD

**Discussion by NicLogix** [celsworth/lxp-bridge#179] [Q&A] [answered]
I must be missing something, but where do I find the current consumption (EPS Output)?

I am using a LUX SNP5000PWV, with the Home Assistant addon.

Publushed on inputs/all:
  "status": 64,
  "v_pv_1": 0,
  "v_pv_2": 0,
  "v_pv_3": 385.8,
  "v_bat": 52.6,
  "soc": 86,
  "soh": 0,
  "p_pv": 0,
  "p_pv_1": 0,
  "p_pv_2": 0,
  "p_pv_3": 0,
  "p_charge": 0,
  "p_discharge": 455,
  "v_ac_r": 0,
  "v_ac_s": 0,
  "v_ac_t": 1433.7,
  "f_ac": 0,
  "p_inv": 0,
  "p_rec": 0,
  "pf": 0,
  "v_eps_r": 230.3,
  "v_eps_s": 1004.8,
  "v_eps_t": 417,
  "f_eps": 49.99,
  "p_eps": 420,
  "s_eps": 507,
  "p_to_grid": 0,
  "p_to_user": 0,
  "e_pv_day": 0,
  "e_pv_day_1": 0,
  "e_pv_day_2": 0,
  "e_pv_day_3": 0,
  "e_inv_day": 0,
  "e_rec_day": 0.6,
  "e_chg_day": 0,
  "e_dischg_day": 0.9,
  "e_eps_day": 0.7,
  "e_to_grid_day": 0,
  "e_to_user_day": 3.9,
  "v_bus_1": 386.9,
  "v_bus_2": 367,
  "e_pv_all": 0,
  "e_pv_all_1": 0,
  "e_pv_all_2": 0,
  "e_pv_all_3": 0,
  "e_inv_all": 0,
  "e_rec_all": 6.7,
  "e_chg_all": 3.5,
  "e_dischg_all": 3.1,
  "e_eps_all": 2.1,
  "e_to_grid_all": 0,
  "e_to_user_all": 30.7,
  "t_inner": 23,
  "t_rad_1": 20,
  "t_rad_2": 23,
  "t_bat": 22,
  "runtime": 239852,
  "max_chg_curr": 50,
  "max_dischg_curr": 95,
  "charge_volt_ref": 57.4,
  "dischg_cut_volt": 0,
  "bat_status_0": 0,
  "bat_status_1": 0,
  "bat_status_2": -16384,
  "bat_status_3": 0,
  "bat_status_4": 0,
  "bat_status_5": 4608,
  "bat_status_6": -21569,
  "bat_status_7": 4,
  "bat_status_8": 24604,
  "bat_status_9": 4253,
  "bat_status_inv": 3,
  "bat_count": 1,
  "bat_capacity": 50,
  "bat_current": -10,
  "bms_event_1": 0,
  "bms_event_2": 0,
  "max_cell_voltage": 0,
  "min_cell_voltage": 0,
  "max_cell_temp": 0,
  "min_cell_temp": 0,
  "bms_fw_update_state": 523,
  "cycle_count": 3,
  "vbat_inv": 52.3,
  "time": 1689677161,
  "datalog": "BAxxxxxxx"

Thank you.

**Comment by celsworth** [comment, marked as answer]
`p_eps` is the EPS output in watts - currently showing 420W in your paste (`s_eps` is in VA but you probably want `p_`). And for completeness, `f_eps` is the frequency.

**Reply by NicLogix** [reply]
Thank you. ( sometimes I think too technical, imagining values on different transformer stages or three phase ratings when i see r s t)
And for those who wonder:
VA is Volt Amps.  Total output, Apparent power.
Watts (in these values) is Reactive power.
The ratio between the two in % is the power factor.

**Reply by celsworth** [reply]
👍 

Yep there is some 3-phase here, for example:

```
"v_ac_r": 0,
"v_ac_s": 0,
"v_ac_t": 1433.7,
```

Although I don't think this is available for EPS, I guess thats always just one phase?

The prefixes should all be consistent in our JSON output, so:

```
p_ = power in watts
s_ = power in VA
f_ = frequency, Hz
v_ = volts
e_ = energy, Watt-hours
t_ = temperature, Celsius
```

**Comment by NicLogix** [comment]
Just to get back to the Consumption Output, I see this is a calculation done by inverter.
To combine all modes, and have single entity, I sum them as follows:
Consumption(W)  = p_to_user + p_eps + p_inv - p_rec
This value calculates exactly the same as on inverter and app.
On a 5KW, ( Consumption/5000 )*100 will give the consumption in %.
